packages/outline-docs/src/guides/component-development/06-properties.mdx

```diff
@@ -21,27 +21,34 @@ import '@phase2/outline-container';
   bottom-margin="spacing-8"
 >
   ## Description
-  Lit components receive input and store their state as JavaScript class fields or properties.
+  Outline components receive input and store their state as JavaScript class fields(normally called attributes) or properties.
   
   [Reactive properties](https://lit.dev/docs/components/properties/) are properties that can trigger the reactive update cycle when changed, re-rendering the component, and optionally be read or written to attributes.
 
   ## Documentation Status
-  <outline-alert status="warning" size="large">
-    <span slot="header">Status: Needs Work/In Progress</span>
+  <outline-alert status="success" size="large">
+    <span slot="header">Status: Complete / Update to data May 2023</span>
     <p>
       This documentation is in need of additional work, or is currently in progress.
     </p>
   </outline-alert>
 </outline-container>
 
 ### Property usage basics
-
 - Properties should be commented using [JSDoc](https://github.com/jsdoc/jsdoc).
 - Each property should provide a comment including a description of the attribute functionality or purpose.
 - Pass to `type` the appropriate [converter](https://lit.dev/docs/components/properties/#conversion-type) (`String`, `Boolean`, etc.).
 - Pass to `attribute` an attribute name in kebab-case.
 - Attribute names must not collide with HTML standard ones `title`, `href`, etc.
   - This means it is best to name properties more like `link-title`, `link-href`, etc.
+- Lit [documentation]()
+
+### Slots versus properties
+- When do I use a property? 
+  - Properties are used for non-content features that impact the state of the component. For example, active versus inactive states.
+- The component shouldn't change its own public properties, except in response to user input. For example, you could set a property to change background color, but the component itself shouldn't change it's own color.
```

Comment:
I don't know that I agree with this statement entirely. I agree that the most common case for a component to update a reflected attribute would be based on some user action, but I think there are definitely cases where a component is doing something internally that warrants reflecting its internal state back in the HTML markup regardless of any user action.

Using the background color as an example, what if your component wants to go look up the browser configuration for dark mode and modify the background color that it uses in some way? Or override the default color with a user preference setting it pulls from an API endpoint?

A more realistic example would be timed carousel with a property the reflects the currently active slide index.

I feel like a better callout might be to only use `reflect: true` on a property when the component will actually modify that property internally, which is a pattern that many components in the repo are already failing to follow correctly:
https://github.com/search?q=repo%3Aphase2%2Foutline+%22reflect%3A+true%22&type=code

https://lit.dev/docs/components/properties/#reflected-attributes

Reply:
I agree with this. A lot of the original components, IIRC had `reflect: true` on by default as "test mode", and it was argued that it was valuable to have that. 

I agree it shouldn't be the default, but used when necessary and appropriate.